src/main/java/edu/cornell/kfs/module/cam/rest/resource/CuCapAssetInventoryApiResource.java

```diff
@@ -131,6 +144,42 @@ public Response updateAsset(@PathParam(CuCamsConstants.CapAssetApi.CAPITAL_ASSET
         }
     }
 
+    private void createCapitalAssetErrorDocument(String netid, String assetTag, String condition, String buildingCode, String roomNumber) {
+        try {
+            GlobalVariables.setUserSession(new UserSession(CuCamsConstants.CapAssetApi.KFS_SYSTEM_USER));
```

Comment:
Instead of explicitly changing the current UserSession in this manner, I would suggest using the GlobalVariables.doInNewGlobalVariables() method instead. This helper method will handle the GlobalVariables management automatically; you just have to pass in the new UserSession and a lambda expression (or method reference) for the code you want to run.